src/components/plotly/shared/custom-plot.js

```diff
@@ -52,25 +53,45 @@ const CustomPlot = ({
 
   // determine the keys for the legend
   const legendKeys = useMemo(() => plotlyInterfaces[type].getLegendKeys(data), [data, type])
-
+  
   // use customColors or generate the color scheme based on a single base color
-  const colors = useMemo(() => customColors.length
-    ? customColors
-    : getColorScheme(baseColor, legendKeys.length)
-  , [customColors, baseColor, legendKeys.length])
+  const colors = useMemo(() => {
+    const customColorsLength = Object.keys(customColors).length
+    const colorOptions = customColorsLength ? customColors : baseColor
+    let numberOfColors = [legendKeys.length]
+
+    if (PLOTLY_MULTI_CHARTS.includes(type) && (
+      Object.keys(baseColor).length > 1 || customColorsLength > 1
+    )) {
+      numberOfColors = [...multiChartLength]
+    }
+
+    return Object.keys(colorOptions).map((key, i) => (
+      customColorsLength ? customColors[key] : getColorScheme(colorOptions[key], numberOfColors[i])
+    ))
+  }, [customColors, baseColor, legendKeys.length, multiChartLength, type])
+
+  const getColors = useCallback(() => {
+    if (PLOTLY_MULTI_CHARTS.includes(type) && colors.length > 1) {
+      const slicedArray = []
+      colors.forEach((c, i) => slicedArray.push(...c.slice(0, multiChartLength[i])))
```

Comment:
We can use here map as well, the way Leo pointed out above.

Reply:
I tried but slice returns an array and I need its content to be spread into one array. For example I have 2 colors with map it returns [ [c1, c2], [c3, c4] ] and what I want is [ c1, c2, c3, c4 ]. map doesn't let me to return `...c.slice(0, multiChartLength[i])` If you know please show me how to use map in this case

Reply:
you can do `map` (which has access to element index on its second param) then flatten with `flat` if you that's the resulting effect you need:

```js
const sliced = colors.map((c, i) => c.slice(0, ...[i])).flat()
```